src/flog.rs

```diff
@@ -149,30 +150,68 @@ pub mod categories {
 /// and let Rust figure it out.
 /// Clients can opt a Debug type into Floggable by implementing FloggableDebug:
 ///    impl FloggableDebug for MyType {}
-pub trait FloggableDisplay {
+pub trait FloggableDisplay: std::fmt::Display {
     /// Return a string representation of this thing.
-    fn to_flog_str(&self) -> String;
+    fn to_flog_str(&self) -> Vec<u8> {
+        self.to_string().into_bytes()
+    }
+}
+
+// Special handling for `WString` to decode PUA codepoints back into the original bytes.
+impl FloggableDisplay for WString {
+    fn to_flog_str(&self) -> Vec<u8> {
+        wcs2string(self)
```

Comment:
> I've used this for debugging. Not sure if we actually want this behavior for normal logging.

printing the original characters seems correct; definitely better than printing weird PUA characters.
I don't see what else "normal logging" could use.. I guess some use cases would prefer to see escaped versions of the bytes (`\xff` etc.) but that's weird because we don't eescape anything else here.